Assets/Scripts/NetworkClient.cs

```diff
@@ -273,4 +312,61 @@ public bool IsConnected()
     {
         return mainWebSocket != null && mainWebSocket.State == WebSocketState.Open;
     }
+
+    private Dictionary<string, string> GetURLQueryParameters()
+    {
+        var output  = new Dictionary<string, string>();
+
+        string url = Application.absoluteURL;
+        if (url?.Length > 0)
+        {
+            var splitUrl = url.Split('?');
+            if (splitUrl.Length == 2)
+            {
+                var queryString = splitUrl[1];
+                var paramsCollection = HttpUtility.ParseQueryString(queryString);
+
+                foreach (var key in paramsCollection.AllKeys)
+                {
+                    output[key] = paramsCollection[key];
+                }
+            }
+        }
+
+        return output;
+    }
+
+    private void GetConnectionParameters(Dictionary<string, string> queryParams,                                   
+                                         out string? _serverHostnameOverride,
+                                         out int? _serverPortOverride)
+    {
+        _serverHostnameOverride = null;
+        _serverPortOverride = null;
+
+        // Override server URL from query arguments.
+        if (queryParams.TryGetValue("server_hostname", out string serverHostnameString))
+        {
+            if (Uri.CheckHostName(serverHostnameString) != UriHostNameType.Unknown)
```

Comment:
This looks handy. What does CheckHostName actually check? Like basic syntax for domain names?

Reply:
It checks for a valid IPv4, IPv6 or DNS name [[ref]](https://learn.microsoft.com/en-us/dotnet/api/system.uri.checkhostname?view=net-8.0).